src/ripple/app/tx/impl/SetAccount.cpp

```diff
@@ -176,6 +176,21 @@ SetAccount::doApply ()
     bool bSetDisallowXRP   = (uTxFlags & tfDisallowXRP) || (uSetFlag == asfDisallowXRP);
     bool bClearDisallowXRP = (uTxFlags & tfAllowXRP) || (uClearFlag == asfDisallowXRP);
 
+    bool sigWithMaster = false;
+
+    {
+        auto const blob = ctx_.tx.getSigningPubKey();
+
+        if (!blob.empty ())
+        {
+            auto const signingPubKey =
+                RippleAddress::createAccountPublic(blob);
+
+            if (calcAccountID(signingPubKey) == account_)
+                sigWithMaster = true;
+        }
```

Comment:
Even if their were `throw`s to worry about, the original code allowed them to propagate to the `catch`es in `preflight/preclaim/doApply`. This modified code has the same mechanism.